### PR TITLE
Update github.com/satori/go.uuid to the latest version

### PR DESCRIPTION
**Pull request by rycus86**
Update the `uuid` library to the latest version, that should have an Insecure Randomness problem https://snyk.io/vuln/SNYK-GOLANG-GITHUBCOMSATORIGOUUID-72488 (https://cwe.mitre.org/data/definitions/338.html) resolved.

Snyk right now flags up projects that transitively depend on docker/licensing.

### CODE REVIEW

**Comment by rycus86** [comment]
Is there anything I can do here to help? cc @dhiltgen

**Comment by rycus86** [comment]
I've fixed the lint issue.
I also had to update the Makefile because the golint package import has changed: https://github.com/golang/lint/issues/415
`go get` now also installed `golint` into my `$GOROOT/bin/` folder, not sure what the new place should be. This was working for me but now I think it might be my setup and will actually break for others.
Would you happen to know?

File: Makefile [hunk 1]
```diff
@@ -16,8 +16,8 @@ vet:
 .PHONY: lint
 lint:
 	@echo "============= Go lint ==============="
-	@go get github.com/golang/lint/golint
-	@test -z "$$(go list ./... | xargs -L1 "$(GOPATH)"/bin/golint | tee /dev/stderr)"
+	@go get golang.org/x/lint/golint
+	@test -z "$$(go list ./... | xargs -L1 golint | tee /dev/stderr)"
```
Comment (rycus86) [inline Makefile]: This path change is what I'm not so sure about...

**Comment by thaJeztah** [comment]
Saw we switched to https://github.com/google/uuid in other repositories (e.g. https://github.com/moby/moby/pull/38868)

@justincormack any preference ?

**Comment by justincormack** [comment]
@thaJeztah yes I prefer to drop the satori version completely, they still haven't managed to get out a release with the fix, it is not maintained.

**Comment by justincormack** [comment]
(the call just needs to be `uuid.New().String()` in the Google version).

Another issue is if you use Go modules it will still choose the latest release, not the fixed head, so everyone is going to revert back to the broken one when the switch to modules.

**Comment by rycus86** [comment]
Thanks for the comments! I'll have a look at switching to Google's version of the library.

**Comment by rycus86** [comment]
See https://github.com/docker/licensing/pull/22 instead.
Should we just close this?

**Comment by mason-fish** [comment]
closed in favor of https://github.com/docker/licensing/pull/22

**Pull request closed without merging**